### PR TITLE
Enable MySQL 8 CI jobs

### LINKED ISSUE #11164: Enable MySQL 8 CI jobs
**Description**

We've never run any CI jobs for MySQL 8 in our CI system. All the jobs are running at 5.7. With the recent refractor, it is as easy as changing CURRENT_MYSQL_VERSIONS to include the MySQL 8 and change the Dockerfile.ci to use mysql8 client as well (currently Dockerfile.ci uses MySQL 5.7 client).

I think this is a hard pre-requisite for the HA Scheduler PR (#9630) to be merged. HA scheduler will only work in a single-scheduler mode for MySQL 5.7 and without running the whole test suite for MySQL 8 we might be giving the users who would likely try MySQL 8 version of Airflow that does not really work well. 

This has some consequences: our build will run quite a bit longer (we might limit versions of python to test with it but still this is at least 20% longer. So it would be best if we implement #10507 and pre-requisite for it is #11163.
Otherwise, our PRs will be blocking each other and we will slow down our own development.

Note that implementing it might be as easy as adding the version and changing client and it **just work** but since we did not ever run it, it is likely there will be some problems to solve.

**Use case / motivation**

When HA scheduler will get in hands of our users as first Alpha release, we have to be quite sure that it works also for MySQL 8 because  only with MySQL 8 users will be able to test the HA setup for their MySQL installations.

**Related Issues**

#10507 #11163

### PR DESCRIPTION
**Pull request by kaxil**
closes https://github.com/apache/airflow/issues/11164

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: scripts/docker/install_mysql.sh [hunk 1]
```diff
@@ -51,6 +51,7 @@ if [[ ${INSTALL_MYSQL_CLIENT:="true"} == "true" ]]; then
     rm -rf "${GNUPGHOME}"
     apt-key list > /dev/null
     echo "deb http://repo.mysql.com/apt/debian/ stretch mysql-5.7" | tee -a /etc/apt/sources.list.d/mysql.list
+    echo "deb http://repo.mysql.com/apt/debian/ stretch mysql-8.0" | tee -a /etc/apt/sources.list.d/mysql.list
```
Comment (potiuk) [inline scripts/docker/install_mysql.sh]: Wy probably want to switch to buster from stretch since we are using buster as base.

Comment (kaxil) [reply scripts/docker/install_mysql.sh]: oh yes

**Comment by potiuk** [comment]
There was a transient server error in the build job. I restarted it.

**Comment by potiuk** [comment]
I believe those tests were cancelled on timeout - I think we have 60 minutes of timeout for the test jobs and some tests could have been hanging there

**Comment by potiuk** [comment]
(commenting from my phone so not sure if that was it)

**Comment by potiuk** [comment]
Maybe changing back to stretch will help? At least to find out the cause? It could also be somy mysql8 client incompatibilities though.

**Comment by kaxil** [comment]
> Maybe changing back to stretch will help? At least to find out the cause? It could also be some mysql8 client incompatibilities though.

Yeah I am going to try that 🤞

**Comment by potiuk** [comment]
FYI. Hey @kaxil - I just updated to buster 5.7 in the v1-10-test branch and it seems to work fine there (for example here: https://github.com/apache/airflow/actions/runs/286961639) so I do not think it is the buster version. I needed to change it there to solve a missing mysqlclient20 library in 1.10 3.5 prod image version in upgrade-checl PRs, Seems that this is fixed now (but I want to confirm that). I also cherry-picked the latest changes for ci/breeze/scripts to v1-10-test.

If MySQL8 still does not work on Monday - seems that I am close to finishing some of my obligations and might return to pure 2.0 Airflow sooner than I thought, but I would like to focus on the separation of the providers, but maybe we could brainstorm on Monday or earlier what could be the reason/fix if you still have problem with it.

File: scripts/docker/install_mysql.sh [hunk 2]
```diff
@@ -50,6 +50,7 @@ if [[ ${INSTALL_MYSQL_CLIENT:="true"} == "true" ]]; then
     gpgconf --kill all
     rm -rf "${GNUPGHOME}"
     apt-key list > /dev/null
+    echo "deb http://repo.mysql.com/apt/debian/ buster mysql-5.7" | tee -a /etc/apt/sources.list.d/mysql.list
```
Comment (potiuk) [inline scripts/docker/install_mysql.sh]: I think MYSQL8 client libs should work with both 5.7 and 8 - MySQL client libraries are meant to be fully backwards compatible - in v1-10-test we have now 5.7 client libraries only and they work fine with both 5.6 and 5.7 versions.

Comment (potiuk) [reply scripts/docker/install_mysql.sh]: But yeah.. Maybe that will help :)

Comment (kaxil) [reply scripts/docker/install_mysql.sh]: Yeah I tried locally, basically, the `libmysqlclient20` was MySQL 5.7 only so the "Build image" step was failing initially. But should work now 🤞 .

I am just trying to keep both but yeah, might remove 5.7 if it is totally compatible

Comment (potiuk) [reply scripts/docker/install_mysql.sh]: I think only one will be used anyway if both are installed with both DBs (not sure which one though)

**Comment by kaxil** [comment]
The following test hangs for MySQL 8

```
tests/providers/mysql/hooks/test_mysql.py::TestMySql::test_mysql_hook_test_bulk_load
```

Need to debug further

**Comment by kaxil** [comment]
@potiuk CI is green

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Cool !

**Pull request merged**